### PR TITLE
Clarify mapping of target env to SPIR-V version

### PR DESCRIPTION
**Pull request by dneto0**
It depends on the API method.

### CODE REVIEW

**Comment by dneto0** [comment]
cc: @johnkslang

File: include/spirv-tools/libspirv.hpp [hunk 1]
```diff
@@ -300,10 +307,19 @@ class SpirvTools {
...
   // registered.
+  // Validates for SPIR-V spec rules for the SPIR-V version named in the binary's
+  // header (at word offset 1).  Additionally, if the target environment is a
+  // client API (such as Vulkan 1.1), then validate for that client API version,
+  // to the extent that it is verifiable from data in the binary itself.
   bool Validate(const std::vector<uint32_t>& binary) const;
   // |binary_size| specifies the number of words in |binary|.
   bool Validate(const uint32_t* binary, size_t binary_size) const;
   // Like the previous overload, but takes an options object.
+  // Validates for SPIR-V spec rules for the SPIR-V version named in the binary's
```
Comment (alan-baker) [inline include/spirv-tools/libspirv.hpp]: A little odd to repeat this here, but not in the previous overload.

Comment (dneto0) [reply include/spirv-tools/libspirv.hpp]: Right. The previous one should say "like the previous one..." as well.  Anyway, I duplicated the text.

**Comment by dneto0** [comment]
> No changes were necessary in optimizer.hpp? The optimizer runs validation by default.

Right, I skipped that.  I added a commit that I think captures the optimizer semantics.  Hey @zoddicus I'm fuzzy on the behaviour of which validations change when setting the environment after optimizer object creation (used for WebGPU to Vulkan and vice versa).  Please verify that the comments are right.

**Comment by zoddicus** [comment]
I assume this blocked by https://github.com/google/shaderc/pull/973?

**Comment by dneto0** [comment]
> I assume this blocked by google/shaderc#973?

No, this is an independent stream.  This is only a doc update to (better) explain handling of target env.

**Pull request merged**